packages/opencensus-exporter-stackdriver/src/stackdriver-stats-utils.ts

```diff
@@ -0,0 +1,288 @@
+   * @param metricDescriptorType The kind of metric.
+   */
+  static createValueType(metricDescriptorType: MetricDescriptorType):
+      ValueType {
+    if (metricDescriptorType === MetricDescriptorType.CUMULATIVE_DOUBLE ||
+        metricDescriptorType === MetricDescriptorType.GAUGE_DOUBLE) {
+      return ValueType.DOUBLE;
+    } else if (
+        metricDescriptorType === MetricDescriptorType.GAUGE_INT64 ||
+        metricDescriptorType === MetricDescriptorType.CUMULATIVE_INT64) {
+      return ValueType.INT64;
+    } else if (
+        metricDescriptorType === MetricDescriptorType.GAUGE_DISTRIBUTION ||
+        metricDescriptorType === MetricDescriptorType.CUMULATIVE_DISTRIBUTION) {
+      return ValueType.DISTRIBUTION;
+    } else {
+      return ValueType.VALUE_TYPE_UNSPECIFIED;
+    }
+  }
+
+  /**
+   * Constructs a LabelDescriptor from a LabelKey.
+   * @param tag The Tags to get TimeSeries information from.
+   */
+  static createLabelDescriptor(labelKeys: LabelKey[]): LabelDescriptor[] {
+    const labelDescriptorList = labelKeys.map(labelKey => {
+      return {
+        key: labelKey.key,
+        valueType: 'STRING',  // Now we only support String type.
+        description: labelKey.description
+      } as LabelDescriptor;
+    });
+
+    // add default "opencensus_task" label.
+    labelDescriptorList.push({
+      key: StackdriverStatsExporterUtils.OPENCENSUS_TASK,
+      valueType: 'STRING',
+      description: StackdriverStatsExporterUtils.OPENCENSUS_TASK_DESCRIPTION
+    });
+    return labelDescriptorList;
+  }
+
+  /**
+   * Creates a Metric using the LabelKeys and LabelValues.
+   * @param metricDescriptor The OpenCensus MetricDescriptor.
+   * @param labelValues The OpenCensus LabelValue.
+   */
+  static createMetric(
+      metricDescriptor: OCMetricDescriptor, labelValues: LabelValue[],
+      metricPrefix: string): {type: string; labels: {[key: string]: string};} {
+    const type = this.getMetricType(metricDescriptor.name, metricPrefix);
+    const labels: {[key: string]: string} = {};
+    for (let i = 0; i < labelValues.length; i++) {
+      const value = labelValues[i].value;
+      if (value && metricDescriptor.labelKeys[i]) {
+        labels[metricDescriptor.labelKeys[i].key] = value;
+      } else {
+        // TODO(mayurkale) : consider to throw an error when LabelValue and
+        // LabelKey lengths are not same.
+      }
+    }
+    labels[StackdriverStatsExporterUtils.OPENCENSUS_TASK] =
+        StackdriverStatsExporterUtils.OPENCENSUS_TASK_VALUE_DEFAULT;
+    return {type, labels};
+  }
+
+  /**
+   * Converts timeseries's point, so that metric can be uploaded to StackDriver.
+   * @param point The timestamped measurement.
+   * @param startTimeStamp The start timestamp of timeseries.
+   * @param valueType The ValueType of metric.
+   */
+  static createPoint(
+      point: TimeSeriesPoint, startTimeStamp: Timestamp,
+      valueType: ValueType): Point {
+    let value;
```

Comment:
Optional: you can avoid use of `let` here by creating a separate function like `createValue` or similar that would take in the value type and value and return the value, then just have `const value = createValue(valueType, point)`